src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/failover/FailoverConnectionPlugin.java

```diff
@@ -410,10 +410,9 @@ private void attemptConnectionUsingCachedTopology() throws SQLException {
 
     metricsContainer.registerUseCachedTopology(true);
 
-    int candidateIndex = getCandidateIndexForInitialConnection();
-    if (candidateIndex != NO_CONNECTION_INDEX) {
-      connectTo(candidateIndex);
-    }
+    // Removed code to call getCandidateIndexForInitialConnection(), and connect to it. This is because on initial
```

Comment:
This method is called `attemptConnectionUsingCachedTopology`, and this method is called in `createConnection`. If we remove this piece of code it changes the intent of `attemptConnectionUsingCachedTopology`.
Do we still need `attemptConnectionusingCachedTopology`? And do we need to modify the code that is using this function?

Reply:
Good catch! I think we should remove the method as it also has the effect of updating the FailoverConnectionPlugin hosts with the cached hosts. But rename it seems like a good idea.

Reply:
I just noticed that this also leaves some unused private methods. I'll remove those as well.

Reply:
Update: Leaving in the method, but with the name changed to reflect what the method actually does now. Also removed SqlException from the signature, and try catch from the user of this method. .